src/main/resources/assets/computercraft/lua/rom/programs/http/gist.lua

```diff
@@ -0,0 +1,669 @@
+    if id:find("/") ~= nil then id, file = id:match("^([0-9A-Fa-f:]+)/(.+)$") end
+    if id:find(":") ~= nil then id = id:gsub(":", "/") end
+    write("Connecting to api.github.com... ")
+    local handle = http.get("https://api.github.com/gists/" .. id)
+    if handle == nil then print("Failed.") return nil end
+    if handle.getResponseCode() ~= 200 then print("Failed.") handle.close() return nil end
+    local meta = json.decode(handle.readAll())
+    handle.close()
+    if meta == nil or meta.files == nil then print("Failed.") return nil end
+    print("Success.")
+    if file then return getGistFile(meta.files[file]), file
+    elseif next(meta.files, next(meta.files)) == nil then return getGistFile(meta.files[next(meta.files)]), next(meta.files)
+    elseif meta.files["init.lua"] ~= nil then return getGistFile(meta.files["init.lua"]), "init.lua"
+    else
+        local luaFiles = {}
+        for k in pairs(meta.files) do if k:match("%.lua$") then table.insert(luaFiles, k) end end
+        table.sort(luaFiles)
+        if #luaFiles == 0 then
+            print("Error: Could not find any Lua files to download!")
+            return nil
+        end
+        if #luaFiles > 1 then print("Warning: More than one Lua file detected, downloading the first one alphabetically.") end
+        return getGistFile(meta.files[luaFiles[1]]), luaFiles[1]
+    end
+end
+
+local function setTextColor(c) if term.isColor() then term.setTextColor(c) elseif c == colors.white then term.setTextColor(c) else term.setTextColor(colors.lightGray) end end
+
+local function requestAuth(headers)
+    if settings.get("gist.id") ~= nil then headers.Authorization = "token " .. settings.get("gist.id") else
+        setTextColor(colors.yellow)
+        write("You need to add a Personal Access Token (PAK) to upload Gists. Follow the instructions at ")
+        setTextColor(colors.blue)
+        write("https://tinyurl.com/GitHubPAK")
+        setTextColor(colors.yellow)
+        write(" to generate one. Make sure to check the '")
+        setTextColor(colors.blue)
+        write("gist")
+        setTextColor(colors.yellow)
+        print("' checkbox on step 7 (under 'Select scopes'). Once done, paste it here.")
+        setTextColor(colors.lime)
+        write("PAK: ")
+        setTextColor(colors.white)
+        local pak = read()
+        if pak == nil or pak == "" then error("Invalid PAK, please try again.") end
+        settings.set("gist.id", pak)
+        headers.Authorization = "token " .. pak
+    end
+end
+
+local args = {...}
+
+local helpstr = "Usages:\ngist put <filenames...> [-- description...]\ngist edit <id> <filenames...> [-- description]\ngist delete <id>\ngist get <id> <filename>\ngist run <id> [arguments...]\ngist info <id>"
+
+if #args < 2 then
+    print(helpstr)
+    return 1
+end
+
+if not http then
+    printError("Gist requires http API")
+    if _G.config ~= nil then printError("Set http_enable to true in the CraftOS-PC configuration")
```

Comment:
Hrm, might just be better to say "ComputerCraft's configuration" - the filename changes between versions. I'll update pastebin and wget to be the same.